### ISSUE TITLE
OSX backend issues and fix suggestion (IME and window positions)

### ISSUE THREAD

**Issue by 734vin** [ocornut/imgui#7028]
## info

Version: 1.90.0
Branch: docking

Back-ends: imgui_impl_osx.cpp + imgui_impl_metal.cpp
Compiler: xcode(clang)
Operating System: macos 13.5

## issues

1. ime position

ime postion is always wrong.
this function in imgui_impl_osx.cpp
```
- (void)updateImePosWithView:(NSView *)view
{
    NSWindow *window = view.window;
    if (!window)
        return;
    //NSRect contentRect = [window contentRectForFrameRect:window.frame];
    //NSRect rect = NSMakeRect(_posX, contentRect.size.height - _posY, 0, 0);
    //_imeRect = [window convertRectToScreen:rect];
    
    //-------------------------------------------------- fix
    NSRect contentRect = window.contentLayoutRect;
    //no tile bar window should be considered.
    if(window.styleMask & NSWindowStyleMaskFullSizeContentView ) contentRect = window.frame;
    _imeRect = NSMakeRect(_posX, window.screen.frame.size.height - _posY, 0, 0); 
    
    //multiple screens should be considered, postion of ime in system is relative to main screen.
    NSScreen* mainScreen = NSScreen.screens.firstObject;
    if(window.screen != mainScreen)  
    {
        //_imeRect.origin.x += window.screen.frame.origin.x - mainScreen.frame.origin.x;
        _imeRect.origin.y += window.screen.frame.origin.y - mainScreen.frame.origin.y;
    }

    //dont need covert to screen position
}
```

2. window postion

this function in imgui_impl_osx.cpp
```
static ImVec2 ImGui_ImplOSX_GetWindowPos(ImGuiViewport* viewport)
{
    ImGuiViewportDataOSX* data = (ImGuiViewportDataOSX*)viewport->PlatformUserData;
    IM_ASSERT(data->Window != 0);

    NSWindow* window = data->Window;
    NSScreen* screen = window.screen;
    NSSize size = screen.frame.size;
    NSRect frame = window.frame;
    NSRect rect = window.contentLayoutRect;
    //return ImVec2(frame.origin.x, size.height - frame.origin.y - rect.size.height);
    
    //-------------------------------------------------- fix
    NSScreen* mainScreen = NSScreen.screens.firstObject;
    NSSize mainScreenSize = size;
    
    //window.frame is relative to main screen. multiple screens should be considered.
    if(mainScreen) mainScreenSize = mainScreen.frame.size; 
    
    //no tile bar window should be considered.
    if(window.styleMask & NSWindowStyleMaskFullSizeContentView ) return ImVec2(frame.origin.x, mainScreenSize.height - frame.origin.y - frame.size.height);
    else return ImVec2(frame.origin.x, mainScreenSize.height - frame.origin.y - rect.size.height);
}
```

## suggestion
all issues fix maybe a suggestion, after all, extensive testing is required.

**Comment by ocornut** [comment]
Thanks for those.
Could someone with access to OSX double check those?
Need to make sure it works with and without multi-viewports (where coordinate space is altered).

**Comment by ocornut** [comment]
I have applied the IME fix with 8deb1007.
Your version has one remaining bug + also this one works when multi-viewports is disabled.
Tested with horizontal and vertical monitor settings.

**Comment by ocornut** [comment]
Applied the GetWindowPos() fix as part of a683033e which attempts to fix a few things together.
OSX backend seems to now work decently with multi-viewport multi-monitors, apart from dpi/scaling issues.

Pretty sure decorations offset are incorrect however:
- if you enable `io.ConfigViewportsNoDecoration` you can notice window-size is not reported correctly.
- Even without `io.ConfigViewportsNoDecoration` it feels like there is a slight offset which I assume may be related to OS window border.
